components/datetime/src/provider/mod.rs

```diff
@@ -193,5 +207,380 @@ pub mod gregory {
+            /// deserialized here into the SkeletonFieldsV1 format which is used in memory
+            /// when working with formatting date times.
+            fn visit_str<E>(self, skeleton_string: &str) -> Result<Self::Value, E>
+            where
+                E: de::Error,
+            {
+                SkeletonFieldsV1::try_from(skeleton_string).map_err(|err| match err {
+                    SkeletonFieldsV1Error::SymbolUnknown(byte) => E::custom(format!(
+                        "Skeleton {:?} contained an unknown symbol, {:?}",
+                        skeleton_string, byte as char
+                    )),
+                    SkeletonFieldsV1Error::SymbolUnimplemented(byte) => E::custom(format!(
+                        "Skeleton {:?} contained an unimplemented symbol, {:?}",
+                        skeleton_string, byte as char
+                    )),
+                    SkeletonFieldsV1Error::FieldTooLong => E::custom(format!(
+                        "Skeleton \"{}\" contained a field that was too long.",
+                        skeleton_string
+                    )),
+                    SkeletonFieldsV1Error::SymbolInvalid(ch) => E::custom(format!(
+                        "Skeleton {:?} contained an invalid symbol, {:?}",
+                        skeleton_string, ch
+                    )),
+                })
+            }
+        }
+
+        #[cfg(feature = "provider_serde")]
+        impl<'de> Deserialize<'de> for SkeletonFieldsV1 {
+            fn deserialize<D>(deserializer: D) -> Result<SkeletonFieldsV1, D::Error>
+            where
+                D: Deserializer<'de>,
+            {
+                deserializer.deserialize_str(SkeletonFieldsDeserializeVisitor)
+            }
+        }
+
+        #[cfg(feature = "provider_serde")]
+        impl Serialize for SkeletonFieldsV1 {
+            fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+            where
+                S: ser::Serializer,
+            {
+                let mut string = String::new();
+
+                for field in self.0.iter() {
+                    let ch: char = field.symbol.into();
+                    for _ in 0..field.length as usize {
+                        string.push(ch);
+                    }
+                }
+
+                serializer.serialize_str(&string)
+            }
+        }
+
+        #[derive(Debug, PartialEq, Clone, Default)]
+        #[cfg_attr(
+            feature = "provider_serde",
+            derive(serde::Serialize, serde::Deserialize)
+        )]
+        pub struct SkeletonTupleV1(pub SkeletonFieldsV1, pub PatternV1);
+
+        #[derive(Debug, PartialEq, Clone, Default)]
+        #[cfg_attr(
+            feature = "provider_serde",
+            derive(serde::Serialize, serde::Deserialize)
+        )]
+        pub struct DateTimeFormatsV1 {
+            pub style_patterns: StylePatternsV1,
+            pub skeletons: Vec<SkeletonTupleV1>,
```

Comment:
tuple_vec_map does not appear to work when you've wrapped the tuple in another struct.